packages/@lwc/engine-dom/src/apis/is-node-from-template.ts

```diff
@@ -25,12 +25,18 @@ export function isNodeFromTemplate(node: Node): boolean {
     if (node instanceof ShadowRoot) {
         return false;
     }
+
+    const rootNode = node.getRootNode();
+    if (rootNode instanceof ShadowRoot && isFalse('synthetic' in rootNode)) {
```

Comment:
`rootNode instanceof ShadowRoot` is also used in the return clause, maybe just storing the value in a const, and using that value in this condition will reduce the number of instanceof operations in this method.

Reply:
`'synthetic' in rootNode` since too loose... maybe `hasOwnProperty` instead?

Reply:
Based on the jsdoc, `isNodeFromTemplate` returns true if a node is rendered by an LWC component. Checking if a node is within a shadow tree is not a sufficient condition. We would also need to check if the host element is associated with an LWC component.

I checked this code for a while now, but what is the expected behavior for light DOM?

Reply:
No @pmdartus, as I explained in the associated ticket, the name of the method is wrong, this is not about templates, this is about shadows. The suggested name was `isNodeShadowed`.